src/clients/checkCredit.js

```diff
@@ -1,16 +1,29 @@
-import Credit from "../models/credit.js";
+import { mainCredit, reserveCredit } from "../models/credit.js";
+
+let mainDatabase = mainCredit
+let secondaryDatabase = reserveCredit
 
 export default async (messageCost) => {
 
     try {
-        const existingCredit = await Credit.findOne()
+        const existingCreditOnMain = await mainDatabase.findOne()
 
-        if ((existingCredit) && (existingCredit.amount - messageCost >= 0)) {
+        if ((existingCreditOnMain) && (existingCreditOnMain.amount - messageCost >= 0)) {
             return true
         } else {
             return false
         }
-    } catch (err) {
-        console.log("Error while checking balance", err);
+    } catch {
+        console.log("Error while checking balance on main db");
+        try {
+            const existingCreditOnSeconday = await secondaryDatabase.findOne()
+            if ((existingCreditOnSeconday) && (existingCreditOnSeconday.amount - messageCost >= 0)) {
+                return true
+            } else {
+                return false
+            }
+        } catch (err) {
+            throw new Error(("something went wrong. unable to check balance on either database. message aborted"))
```

Comment:
Too nested, you could use findOneAndUpdate right of the bat. This can be refactored :)